drizzlepac/haputils/mvm_quality_analysis.py

```diff
@@ -247,6 +266,740 @@ def run_quality_analysis(total_obj_list,
+    """identify sources found in both sourcelists using stsci.stimage.xyxymatch
+
+    Parameters
+    ----------
+    overlap_info : dict
+        dictionary containing all relevant information about a single region of overlapping observations.
+
+    svm_sourcelist_list : list
+        two-element list of astropy.table objects containing the SVM catalogs to crossmatch
+
+    log_level : int, optional
+        The desired level of verboseness in the log statements displayed on the screen and written to the
+        .log file.  Default value is 'NOTSET'.
+
+    Returns
+    -------
+    ref_index : int
+        indicates which of the two svm sourcelists was used as the reference for crossmatch purposes. A value
+        of '0' indicates the sourcelist that is the first element of svm_sourcelist_list was the reference,
+        and a value of '1' indicates that the second svm_sourcelist_list sourcelist was used as the reference.
+
+    comp_index : int
+        indicates which of the two svm sourcelists was used as the comparison for crossmatch purposes. A value
+        of '0' indicates the sourcelist that is the first element of svm_sourcelist_list was the comparison,
+        and a value of '1' indicates that the second svm_sourcelist_list sourcelist was used as the
+        comparison.
+
+    matched_lines_ref : list
+        indices of the reference catalog that crossmatch to the corresponding element in matched_lines_comp.
+
+    matched_lines_ref : list
+        indices of the comparison catalog that crossmatch to the corresponding element in matched_lines_ref.
+
+    cross_match_details : list
+        three-element list containing the number suitable ref and comp sources available for crossmatch, and
+        the number of sources actually crossmatched
+    """
+    log.setLevel(log_level)
+    # The sourcelist with more sources is automatically set as the "reference" catalog that the other
+    # sourcelist (aka the "comparison" catalog) will be crossmatched to
+    if len(svm_sourcelist_list[0]) >= len(svm_sourcelist_list[1]):
+        ref_index = 0
+        comp_index = 1
+    else:
+        ref_index = 1
+        comp_index = 0
+
+    log.info("Reference catalog: {}, {} sources".format(os.path.basename(overlap_info["svm_sourcelist_{}".format(str(ref_index))]), len(svm_sourcelist_list[ref_index])))
+    log.info("Comparison catalog: {}, {} sources".format(os.path.basename(overlap_info["svm_sourcelist_{}".format(str(comp_index))]), len(svm_sourcelist_list[comp_index])))
+
+    # prepare xyxymatch inputs
+    ref_xy = np.stack((svm_sourcelist_list[ref_index]['X-Skycell'], svm_sourcelist_list[ref_index]['Y-Skycell']), axis=1)
+    comp_xy = np.stack((svm_sourcelist_list[comp_index]['X-Skycell'], svm_sourcelist_list[comp_index]['Y-Skycell']), axis=1)
+    
+    # execute crossmatch
+    matches = xyxymatch(comp_xy, ref_xy, tolerance=5.0, separation=1.0)
```

Comment:
You may want to wrap this call in a try...except block since this C-code can fail spectacularly under the right conditions.  Also, if the offsets are more than 5 pixels apart, then it will fail to find a cross-match.  Increasing the tolerance to a larger value to something more like 20pixels does increase the risk of cross-match confusion in crowded fields, but the RMS for the non-GAIA catalogs can actually approach a full-arcsecond compared to GAIA.  Thus, this low tolerance value would actually be blind to such large offsets which are what we are hoping to flag with this analysis.  A try...except block would at least allow this code to continue if it does fail in extremely crowded fields.